core/src/main/scala/org/apache/spark/crypto/CryptoStreamUtils.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.crypto
+
+import java.io.{InputStream, OutputStream}
+import java.util.Properties
+import javax.crypto.spec.{IvParameterSpec, SecretKeySpec}
+
+import org.apache.commons.crypto.random._
+import org.apache.commons.crypto.stream._
+
+import org.apache.spark.SparkConf
+import org.apache.spark.crypto.CryptoConf._
+import org.apache.spark.deploy.SparkHadoopUtil
+
+/**
+ * A util class for manipulating file shuffle encryption and decryption streams.
+ */
+private[spark] object CryptoStreamUtils {
+  // The initialization vector length in bytes.
+  val IV_LENGTH_IN_BYTES = 16
+  // The prefix of Crypto related configurations in Spark configuration.
+  val SPARK_COMMONS_CRYPTO_CONF_PREFIX = "spark.commons.crypto."
```

Comment:
This name comes from the ```Apache Commons Crypto```. User can pass ```Apache Commons Crypto``` related configurations along with prefix "spark." to the crypto stream to be created.